src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -250,7 +285,7 @@ private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableAr
 
         private static bool IsNullOrEmptyStringConstant(BoundExpression operand)
         {
-            return (operand.ConstantValueOpt != null && string.IsNullOrEmpty(operand.ConstantValueOpt.StringValue)) ||
+            return (operand.ConstantValueOpt?.IsString == true && string.IsNullOrEmpty(operand.ConstantValueOpt.StringValue)) ||
```

Comment:
>operand.ConstantValueOpt.StringValue [](http://example.com/codeflow?start=87&length=36)

Is this value ever null?

Reply:
Not sure. This change was made just to make sure operand is of string type since this helper can now get char/ROS as well. This was done pretty early so I don't remember all details, but the change seems pretty conservative. We previously checked for `null`, now we check for `null` and `IsString` at the same time. What do you think can be improved here then?